zipline/pipeline/loaders/blaze.py

```diff
@@ -0,0 +1,870 @@
+        with the name of ``expr`` + '_deltas'. If None is passed, no deltas
+        will be used.
+    loader : BlazeLoader, optional
+        The blaze loader to attach this pipeline dataset to. If None is passed,
+        the global blaze loader is used.
+    resources : dict or any, optional
+        The data to execute the blaze expressions against. This is used as the
+        scope for ``bz.compute``.
+    no_deltas_rule : {'warn', 'raise', 'ignore'}
+        What should happen if ``deltas='auto'`` but no deltas can be found.
+        'log' says to log a message but continue.
+        'raise' says to raise an exception if no deltas can be found.
+        'ignore' says take no action and proceed with no deltas.
+
+    Returns
+    -------
+    pipeline_api_obj : DataSet or BoundColumn
+        Either a new dataset or bound column based on the shape of the expr
+        passed in. If a table shaped expression is passed, this will return
+        a ``DataSet`` that represents the whole table. If an array-like shape
+        is passed, a ``BoundColumn`` on the dataset that would be constructed
+        from passing the parent is returned.
+    """
+    deltas = _get_deltas(expr, deltas, no_deltas_rule)
+    if deltas is not None:
+        invalid_nodes = tuple(filter(is_invalid_deltas_node, expr._subterms()))
+        if invalid_nodes:
+            raise TypeError(
+                'expression with deltas may only contain (%s) nodes,'
+                " found: %s" % (
+                    ', '.join(map(getname, valid_deltas_node_types)),
+                    ', '.join(map(compose(getname, type), invalid_nodes)),
+                ),
+            )
+
+    # Check if this is a single column out of a dataset.
+    single_column = None
+    if bz.ndim(expr) != 1:
+        raise TypeError(
+            'expression was not tabular or array-like,'
+            ' too many dimensions: %d' % bz.ndim(expr)
+        )
+    if isscalar(expr.dshape.measure):
+        # This is a single column. Record which column we are to return
+        # but create the entire dataset.
+        single_column = rename = expr._name
+        field_hit = False
+        if not isinstance(expr, traversable_nodes):
+            raise TypeError(
+                "expression '%s' was array-like but not a simple field of"
+                " some larger table" % str(expr),
+            )
+        while isinstance(expr, traversable_nodes):
+            if isinstance(expr, bz.expr.Field):
+                if not field_hit:
+                    field_hit = True
+                else:
+                    break
+            rename = expr._name
+            expr = expr._child
+        expr = expr.relabel({rename: single_column})
+
+    measure = expr.dshape.measure
+    if not isrecord(measure) or AD_FIELD_NAME not in measure.names:
+        raise TypeError(
+            "expr must be a collection of records with at least an '{ad}'"
```

Comment:
Yeah, I'd bind the name `dataset_expr` at the end of the `while` loop and work with that for the rest of the function.